models/git/branch.go

```diff
@@ -115,6 +118,22 @@ type Branch struct {
 	UpdatedUnix   timeutil.TimeStamp `xorm:"updated"`
 }
 
+// TableCollations git's ref-name is case-sensitive internally, however, in some databases (mssql, mysql, by default), it's case-insensitive at the moment
+// so we need to set the collation to case-sensitive which is utf8mb4_bin for mysql and Latin1_General_CS_AS for mssql
+// the function is supported by xorm
+func (b *Branch) TableCollations() []*schemas.Collation {
+	if setting.Database.Type.IsMySQL() {
+		return []*schemas.Collation{
+			{Name: "utf8mb4_bin", Column: "name"},
+		}
+	} else if setting.Database.Type.IsMSSQL() {
+		return []*schemas.Collation{
+			{Name: "Latin1_General_CS_AS", Column: "name"},
```

Comment:
IIRC SQLServer doesn't treat this collation as UTF-8 internally.

So it's better to add a test for it to make sure wide UTF-8 chars (eg: emoji) can be inserted and selected correctly.

Reply:
Also should check MariaDB as I recall they have different collation names than MySQL and some are not present on both.